Lib/dataclasses.py

```diff
@@ -617,10 +617,10 @@ def _frozen_get_del_attr(cls, fields, globals):
     locals = {'cls': cls,
               'FrozenInstanceError': FrozenInstanceError}
     if fields:
-        fields_str = '(' + ','.join(repr(f.name) for f in fields) + ',)'
+        fields_str = '{' + ','.join(repr(f.name) for f in fields) + '}'
     else:
-        # Special case for the zero-length tuple.
-        fields_str = '()'
+        # Special case for the zero-length set.
+        fields_str = 'set()'
```

Comment:
This call wouldn’t be constant-foldable, so it’s probably a regression. Since `field_str` is being formatted into the one line twice below, if that was moved up here we could instead just eliminate the second clause entirely in this case.

Reply:
Make sense. Thanks for the advice.